pkg/capabilities/capabilities_test.go

```diff
@@ -48,10 +41,58 @@ func TestGet(t *testing.T) {
 			HostNetworkSources: []string{"A", "B"},
 		},
 	}
-	SetForTests(cap)
+	ResetForTest()
+	Initialize(cap)
 
 	res = Get()
 	if !reflect.DeepEqual(cap, res) {
 		t.Fatalf("expected Capabilities: %#v , got a different: %#v", cap, res)
 	}
 }
+func TestSetup(t *testing.T) {
+	testCases := []struct {
+		name                     string
+		allowPrivileged          bool
+		perConnectionBytesPerSec int64
+		expectedCapabilities     Capabilities
+	}{
+		{
+			name:                     "AllowPrivileged true with bandwidth limit",
+			allowPrivileged:          true,
+			perConnectionBytesPerSec: 1024,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        true,
+				PerConnectionBandwidthLimitBytesPerSec: 1024,
+			},
+		},
+		{
+			name:                     "AllowPrivileged false with higher bandwidth limit",
+			allowPrivileged:          false,
+			perConnectionBytesPerSec: 2048,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        false,
+				PerConnectionBandwidthLimitBytesPerSec: 2048,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ResetForTest()
+
+			Setup(tc.allowPrivileged, tc.perConnectionBytesPerSec)
+			res := Get()
+			if !compareCapabilities(tc.expectedCapabilities, res) {
```

Comment:
should be able to directly compare `tc.expectedCapabilities != res`

Reply:
I ran into an issue when trying this one: the `PrivilegedSources` field is not a comparable type. For now, I have just changed it to use `reflect.DeepEqual` which I think addresses the comment. Happy to change it to something else.